test/morphology/test_skeletonize.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+import torch
+from torch.autograd import gradcheck
+
+from kornia.morphology import skeletonize
+from kornia.testing import assert_close
+
+class TestSkeletonize:
+    def test_smoke(self, device, dtype):
+        kernel = torch.rand(3, 3, device=device, dtype=dtype)
+        assert kernel is not None
+
+    @pytest.mark.parametrize("shape", [(1, 1, 4, 4), (2, 1, 2, 4), (3, 1, 4, 1), (3, 1, 5, 5)])
+    @pytest.mark.parametrize("kernel", [(3, 3), (5, 5)])
+    def test_cardinality(self, device, dtype, shape, kernel):
+        img = torch.ones(shape, device=device, dtype=dtype)
+        krnl = torch.ones(kernel, device=device, dtype=dtype)
+        assert skeletonize(img, krnl).shape == shape
+
+    def test_kernel(self, device, dtype):
+        tensor = torch.tensor([[0.5, 1.0, 0.3], [0.7, 0.3, 0.8], [0.4, 0.9, 0.2]], device=device, dtype=dtype)[
+            None, None, :, :
+        ]
+        kernel = torch.tensor([[0.0, 1.0, 0.0], [1.0, 1.0, 1.0], [0.0, 1.0, 0.0]], device=device, dtype=dtype)
+        expected = torch.tensor([[0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]], device=device, dtype=dtype)[
+            None, None, :, :
+        ]
+        assert_close(skeletonize(tensor, kernel), expected, atol=1e-4, rtol=1e-4)
+
+    def test_exception(self, device, dtype):
+        tensor = torch.ones(1, 1, 3, 4, device=device, dtype=dtype)
+        kernel = torch.ones(3, 3, device=device, dtype=dtype)
+
+        with pytest.raises(TypeError):
+            assert skeletonize([0.0], kernel)
```

Comment:
In these cases, `assert` on `pytest.raises` context isn't necessary. You can match the error message by doing 
```python
       with pytest.raises(TypeError) as errinfo:
           skeletonize([0.0], kernel)
       assert 'THE ERROR MESSAGE' in str(errinfo)
```